core/envelope-sim/src/test/java/fr/sncf/osrd/envelope_sim/AllowanceRangesTests.java

```diff
@@ -78,14 +89,15 @@ public void testRangesOfDifferentTypes() {
         var baseTime1 = maxEffortEnvelope.getTimeBetween(0, rangesTransition);
         var baseTime2 = maxEffortEnvelope.getTimeBetween(rangesTransition, length);
         var totalBaseTime = maxEffortEnvelope.getTotalTime();
-        assertEquals(totalBaseTime, baseTime1 + baseTime2, 2 * TIME_STEP);
+        assertEquals(totalBaseTime, baseTime1 + baseTime2, .001);
```

Comment:
We try to move away from epsilons as hard-coded values, using functions like `arePositionsEqual` in `TrainPhysicsIntegrator`. There's no function (yet) for time equality, but it could be added.